fbpic/particles/particles.py

```diff
@@ -703,26 +753,109 @@ def deposit( self, fld, fieldtype ) :
                         grid[0].Jt, grid[1].Jt,
                         grid[0].Jz, grid[1].Jz,
                         self.cell_idx, self.prefix_sum)
-                elif self.particle_shape == 'linear':
-                    deposit_J_gpu_linear[dim_grid_2d_flat, dim_block_2d_flat](
+                else:
+                    raise ValueError("`particle_shape` should be either \
+                                      'linear' or 'cubic' \
+                                       but is `%s`" % self.particle_shape)
+            else:
+                raise ValueError("`fieldtype` should be either 'J' or \
+                                  'rho', but is `%s`" % fieldtype)
+        # CPU multi-threading version
+        elif self.use_threading:
+            # Register particle chunk size for each thread
+            tx_N = int(self.Ntot/self.nthreads) 
+            tx_chunks = [ tx_N for k in range(self.nthreads) ]
+            tx_chunks[-1] = tx_chunks[-1] + int(self.Ntot%self.nthreads)
```

Comment:
@MKirchen Would you be okay with using an array for `tx_chunks` instead of a list? 
I am worried that a list may be cumbersome for numa, since numba then has to check the type of every single element of the list, each time it is calling a jitted function (as opposed to just once for an array).